app/src/main/res/values/arrays.xml

```diff
@@ -31,12 +31,4 @@
         <item>@string/private_string</item>
     </string-array>
 
-    <string name="resource_scheme">"resource://"</string>
-    <string name="chrome_scheme">"chrome://</string>
-    <string name="about_scheme">"about:"</string>
-    <string-array name="mozac_url_schemes_blocklist" >
```

Comment:
Sometimes when lint says a resource is unused, it's wrong because the code is referenced in a tricky way: this is one of those cases. To figure this out, I used our code search tool, https://searchfox.org/mozilla-mobile/source/, to see where this may be referenced from. I found it was [defined and read](https://searchfox.org/mozilla-mobile/search?q=mozac_url_schemes_blocklist&path=&case=false&regexp=false) in our main application library, `android-components`, that fenix consumes. The android-components repo can be found here: https://github.com/mozilla-mobile/android-components/.

I verified that it was still in use by:
- Checking out a commit without your changes to remove this code
- Adding a call to `SafeUrl.stripUnsafeUrlSchemes` in `HomeActivity.onCreate` (`stripUnsafeUrlSchemes` is the code that accesses the resources; it's called in the code normally but I had difficulty reproducing it that way)
- Starting the debugger with a breakpoint on that method ^
- Verifying the definition from fenix is used in the debugger rather than the one in android-components

The reason Android Lint was unable to detect this error is because it doesn't search an application's libraries to determine if code is in use. And since the resource files get merged, fenix's definition overrides the one from android-components.

---

As for what we should do about this issue, we should:
1. leave the code in the app
2. Explicitly ignore the `UnusedResources` error for `mozac_url_schemes_blocklist` (and the other strings if necessary but it might be smart enough not to)
3. Add a comment to explain why we explicitly ignore the error (i.e. the code that accesses this is defined in ac so the linter can't detect it).

In the future, when removing references to unused resources, it's good to double-check they're not used from `android-components` by using the searchfox tool I mentioned above. Naturally, I didn't expect you to know about this in advance so thanks for understanding!

Reply:
Ok sure, will make the required changed

Reply:
This has been resolved . And thanks! I wil use the code search tool to fix the `unused resources` in subsequent PR